app/models/profile.rb

```diff
@@ -0,0 +1,14 @@
+class Profile < ApplicationRecord
+  belongs_to :user
+
+  # This method generates typed accessors for all active profile fields
+  def self.define_store_accessors!
+    ProfileField.active.find_each do |field|
+      store_attribute :data, field.attribute_name, field.type
```

Comment:
This gives us nicely typed accessors types accessors, including the normal `ActiveRecord` type coercions.